tests/i18n/test_extraction.py

```diff
@@ -127,6 +127,33 @@ def assertNotRecentlyModified(self, path):
         self.assertGreater(delta, 10, "%s wasn't recently modified" % path)
 
 
+class NormalizeLocaleTest(ExtractorTests):
+
+    def test_makemessages_with_normalized_locale(self):
+        """
+        Test the makemessages works with normalized locales in the input.
+        """
+
+        for locale in ('zh-cn', 'zh_cn', 'zh_CN', 'ZH-CN',):
+            out = StringIO()
+
+            with self.subTest():
+                # Test the Normalization of different strings used for locales
+                locale_normalized = normalize_locale(locale, out)
+                self.assertEqual('zh_CN', locale_normalized)
+
+            with self.subTest():
+                if locale in ('zh-CN', 'zh_CN', 'ZH-CN',):
+                    # Test the warning given on normalizing the locale
+                    self.assertIn('locale normalized to zh-cn', out.getvalue())
+                elif locale in ('zh-cn',):
+                    # Locale normalization is not needed
+                    self.assertIn('', out.getvalue())
+
+            management.call_command('makemessages', locale=[locale], verbosity=0)
+            self.assertTrue(os.path.exists('locale/zh_CN/LC_MESSAGES/django.po'))
```

Comment:
These generated files still need to be removed. Look around the tests to see how this has been done before...

Reply:
@pope1ni Correct me if I am wrong, `RunInTmpDirMixin` will remove the files generated during the testing.